### PR TITLE
[Woo POS] M2 UI: Enter POS mode loading screen

### LINKED ISSUE #13143: [Woo POS] M2 UI: Enter POS mode loading screen
When we enter in POS mode, there's a delay while we wait for the data back from network. We need to implement a loading screen (or set of loading screens) while we wait for the POS mode to load.

Image

### PR DESCRIPTION
**Pull request by staskus**
Closes: #13143

## Description

Created POS entry loading view. Eventually, it should be visible when originally entering POS or when reloading the empty view. Therefore, I set the visibility conditions to: itemList.loading and tems.count == 0.

## Solution

1. Created `PointOfSaleLoadingView`. Since this loading view is also used in other places, the intention is to make it reusable in other tasks. 
2. Added `Fonts+WooCommercePOS` for reusable font styles taken from Figma
3. Moved loading view from `ItemListView` to `POSDashbordView`, since the view should appear on top of the content. Loading view within `ItemListView` should only appear with pull to refresh logic.
4. Added basic transition animations

Added tests in another PR to avoid too many PR changes: 

## Testing information

1. Open Woo app
6. Select Menu -> Point of Sale Mode
7. Confirm entry loading view appears according to requirements TfaZ4LUkEwEGrxfnEFzvJj-fi-2710_109469
8. Add items to cart, interact with menu, complete the payment, exit pos
9. Confirm that entry loading doesn't appear anymore

## Screenshots

https://github.com/user-attachments/assets/b288b467-2760-4fed-b153-0cfb363d9a14

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/215">19.7</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleLoadingView.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                size: Layout.progressViewSize,
+                lineWidth: Layout.progressViewLineWidth,
+                lineCap: .butt,
+                circleColor: .wooPurple10,
+                fillColor: .wooPurple50))
+    }
+}
+
+private extension PointOfSaleLoadingView {
+    enum Localization {
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleLoadingView.swift:36]: This likely be moved when reusing this view for other status views.

image

File: WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift [hunk 2]
```diff
@@ -28,6 +28,10 @@ extension Color {
         Color(uiColor: .wooCommercePurple(.shade60))
     }
 
+    static var wooPurple10: Color {
+        Color(red: 207.0 / 255.0, green: 185.0 / 255.0, blue: 246 / 255.0)
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift]: `Color(uiColor: .wooCommercePurple(.shade10))`
We have these woo purple shades in `ColorPalette` assets

File: WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift [hunk 3]
```diff
@@ -36,6 +40,10 @@ extension Color {
         Color(red: 255.0 / 255.0, green: 166.0 / 255.0, blue: 14.0 / 255.0)
     }
 
+    static var wooPurple50: Color {
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift]: `Color(uiColor: .wooCommercePurple(.shade50))`

File: WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift [hunk 4]
```diff
@@ -36,6 +40,10 @@ extension Color {
...
+        Color(red: 127.0 / 255.0, green: 84.0 / 255.0, blue: 179.0 / 255.0)
+
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift]: extra line

**Review by bozidarsevo — APPROVED** [review]

**Review by bozidarsevo — CHANGES_REQUESTED** [review]
Hit the approve instead of comment :D

File: WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift (hunk 2 above)
Comment (staskus) [reply WooCommerce/Classes/POS/Utils/Color+WooCommercePOS.swift]: Thanks, didn't notice that, fixed!

**Review by bozidarsevo — APPROVED** [review]

**Pull request merged**